Connecticut

bot190 opened the pull request:
This is the code developed during the Northern Connecticut district event and shortly afterwards.

Quinn-5 committed:
Fixed constant value

Quinn-5 committed:
Added manual control

Quinn-5 committed:
fixes

jeffaburke committed:
More fixes

jeffaburke committed:
Fixed COpilot left joystick

jeffaburke committed:
fixes

jeffaburke committed:
Major fix

jeffaburke committed:
New button bind

jeffaburke committed:
Changed code

jeffaburke committed:
Trigger fix

Rohan-Sreenivasan committed:
wait command

Rohan-Sreenivasan committed:
pierre changes

Rohan-Sreenivasan committed:
Merge branch 'HEAD' into Connecticut

Rohan-Sreenivasan committed:
NEW Pierre Possesion

Makes the spacing better on pierre. Once again thanks to all the software dudes for making this possible

Rohan-Sreenivasan committed:
vision changes

Rohan-Sreenivasan committed:
More vision changes

jeffaburke committed:
Added LED Code plus some fixes

Rohan-Sreenivasan committed:
LED fix

Rohan-Sreenivasan committed:
more fixes

jeffaburke committed:
Vision Changes

jeffaburke committed:
Merge branch 'Connecticut' into origin/Connecticut

jeffaburke committed:
Merge remote-tracking branch 'origin/Connecticut' into Connecticut

Rohan-Sreenivasan committed:
PID Tuning

we tried some mild pid tuning and it works ok but we have a long way to go for perfection

jeffaburke committed:
Camera on off

Rohan-Sreenivasan committed:
slow turn command

jeffaburke committed:
Changed Value

Quinn-5 committed:
Change saturation and HSV for new LEDs

Quinn-5 committed:
Merge remote-tracking branch 'origin/Connecticut' into Connecticut

Rohan-Sreenivasan committed:
shooter spped change

Rohan-Sreenivasan committed:
Merge remote-tracking branch 'origin/Connecticut' into Connecticut

jeffaburke committed:
Added Retract Intake

jeffaburke committed:
Merge remote-tracking branch 'origin/Connecticut' into Connecticut

Rohan-Sreenivasan committed:
new test image

Rohan-Sreenivasan committed:
Merge remote-tracking branch 'origin/Connecticut' into Connecticut

Quinn-5 committed:
updated hsv and camera setting for long disance

Rohan-Sreenivasan committed:
speed change

Rohan-Sreenivasan committed:
Elbit vision changes

jeffaburke committed:
After Match Lift

jeffaburke committed:
Merge remote-tracking branch 'origin/Connecticut' into Connecticut

Rohan-Sreenivasan committed:
Merge remote-tracking branch 'origin/master' into Connecticut

Quinn-5 committed:
elbit stuff

Quinn-5 committed:
Improving filtering methods, better and more efficient line filtering

Quinn-5 committed:
Merge branch 'HEAD' into Connecticut

Rohan-Sreenivasan committed:
VISION V1

it kinda does stuff

msoucy commented on src/main/java/frc/robot/Robot.java:
```diff
@@ -171,14 +181,14 @@ public void testInit() {
 
     public CommandBase shootAuto() {
-        CommandBase cmd = new SequentialCommandGroup(releaseBalls(3), shooter.spinDown(), drive.driveDistance(20, .5));
+        CommandBase cmd = new SequentialCommandGroup(releaseBalls(3), shooter.spinDown(), drive.driveDistance(40, .5));
```
This is the second time I've seen this exact drive command. Can we make a factory function for it?

msoucy commented on src/main/java/frc/robot/Robot.java:
```diff
@@ -190,10 +200,23 @@ public CommandBase releaseBalls(int ballCount) {
     }
 
+    public CommandBase systemsCheck() {
+        CommandBase cmd = new SequentialCommandGroup(intake.intake(), indexer.pierrePossesionNEW(),
```
Don't name something "new", it's not very descriptive. Either get rid of the old one and use that name (if it's a non breaking change) or come up with a more descriptive name for it

msoucy commented on src/main/java/frc/robot/Robot.java:
```diff
@@ -215,6 +238,29 @@ public CommandBase camToggle() {
+    }
+
+    public CommandBase visionAlignment() {
+        CommandBase cmd = new SequentialCommandGroup(camToggle(), led.visionGreenOn(), drive.visionAlignDegrees(),
```
What is it toggling here?

bot190 replied:
Hrm.. this should probably also be `camOn` or `enableTargeting`

jeffaburke replied:
We did have Cam on, I thought I changed it guess not

msoucy commented on src/main/java/frc/robot/maps/FrancoisMap.java:
```diff
@@ -52,7 +53,7 @@ public SendableSpeedController getRight() {
                 SendableRegistry.add(sendLeader.getEncoder(), "Right Drive");
                 SendableRegistry.enableLiveWindow(sendLeader.getEncoder());
-
+                // return sendLeader;
```
Take us to your leader

bot190 replied:
Remove the commented code here

jeffaburke replied:
We are returning a Mod speed controller here do you want me just to get rid of this comment? NVM

bot190 commented on src/main/java/frc/robot/subsystems/Shooter.java:
```diff
@@ -38,7 +38,7 @@
     // inches
     public final static double TARGET_HEIGHT = 98.25;
-    private final static double MAX_RPM = 5200;
+    // private final static double MAX_RPM = 5200;
```
This should be removed, not commented out.

msoucy commented on src/main/java/frc/robot/subsystems/ControlPanel.java:
```diff
@@ -106,7 +106,7 @@ public ColorStates getTargetColor() {
         CommandBase cmd = new FunctionalCommand(() -> {
         }, () -> {
-            spinnerMotor.set(speed.getAsDouble());
+            spinnerMotor.set(-Math.abs(speed.getAsDouble()));
```
Could we be using the speed controllers built in invert functionality instead of having negatives everywhere?

jeffaburke replied:
yeah, we do need to look at why control panel is always trying to spin... and it looks like the Control Panel Motor is already inverted in Francois map so I'll take that out then remove all the negatives

bot190 commented on src/main/java/frc/robot/subsystems/Lift.java:
```diff
@@ -119,12 +125,24 @@ public CommandBase turnOffBrake() {
     }
 
-    public CommandBase moveTicks(double ticks, double speed) {
+    public CommandBase moveTicks(double ticks, double speed, boolean checkStops) {
```
We're doing this a bit differently now, just need Jeff to push his code.

bot190 commented on src/main/java/frc/robot/subsystems/Led.java:
```diff
@@ -0,0 +1,47 @@
+    }
+
+    public CommandBase visionGreenOn() {
+        CommandBase cmd = new InstantCommand(() -> {
```
These two commands could be simplified by defining a function that sets all of the LEDs to a specific color and then just calling that in each command.

jeffaburke replied:
Yeah RGB or HSV?

bot190 replied:
This is closer. This function should be renamed `visionLedColor`, it should also take `saturation` and `value`. You should create a new factory function that just has `return visionLedColor(120);`. The below `visionLedOff` should be the same, but passing in all `0`s

bot190 commented on src/main/java/frc/robot/subsystems/Led.java:
```diff
@@ -0,0 +1,47 @@
+    AddressableLEDBuffer visionBuffer;
+
+    public Led(LEDMap map) {
+        visionBuffer = new AddressableLEDBuffer(6);
```
The length of the buffer should probably also come from the robot map.

msoucy commented on src/main/java/frc/robot/subsystems/Drive.java:
```diff
@@ -132,4 +141,107 @@ public CommandBase turnDegrees(double degrees, double speed) {
+
+    public CommandBase slowTurn(boolean direction) {
+        CommandBase cmd = new RunCommand(() -> {
+            if (direction == true) {
```
I don't like the use of "direction" as a Boolean here. Either make it the speed itself, or change the name to turning left/turningRight.

Also, you can just say "if (direction)" and "else" in this block - the existing code is very redundant

msoucy commented on src/main/java/frc/robot/subsystems/Drive.java:
```diff
@@ -132,4 +141,107 @@ public CommandBase turnDegrees(double degrees, double speed) {
+
+            }
+        }, this);
+        cmd.setName("turing");
```
Alan Turing?

msoucy commented on src/main/java/frc/robot/subsystems/Drive.java:
```diff
@@ -132,4 +141,107 @@ public CommandBase turnDegrees(double degrees, double speed) {
+
+            @Override
+            public void execute() {
+                if (i % 20 == 0) {
```
I is a static variable in the subsystem. Can't it be an instance variable in this command?

bot190 replied:
Stupid Java question, Can you make instance variables inside anonymous classes? Or does this need to become a full class implementation?

msoucy replied:
Instance variables are fine - we use them extensively in chopshoplib

msoucy commented on src/main/java/frc/robot/subsystems/Drive.java:
```diff
@@ -132,4 +141,107 @@ public CommandBase turnDegrees(double degrees, double speed) {
+    }
+
+    public CommandBase visionAlignDegrees() {
+        CommandBase cmd = new CommandBase() {
```
Make sure to use an initializer block to add the subsystem as a dependency - one of the flaws with command base itself.

msoucy commented on src/main/java/frc/robot/subsystems/Drive.java:
```diff
@@ -132,4 +141,107 @@ public CommandBase turnDegrees(double degrees, double speed) {
+
+            @Override
+            public boolean isFinished() {
+                // TODO Auto-generated method stub
```
Get rid of todo

bot190 commented on src/main/java/frc/robot/subsystems/Indexer.java:
```diff
@@ -138,6 +147,31 @@ public CommandBase pierrePossesion() {
+    // run singulator when front intake is triggered, stop singulator when back
+    // intake IR is clear and until bottom pierre is covered
+    // run pierre when bottom pierre is covered
+    public CommandBase pierrePossesionNEW() {
```
We should get rid of `pierrePossesion` now that we have this one.

msoucy commented on src/main/java/frc/robot/subsystems/Drive.java:
```diff
@@ -132,4 +141,107 @@ public CommandBase turnDegrees(double degrees, double speed) {
+
+    // public CommandBase visionAdjust() {
+
+    // CommandBase cmd = new FunctionalCommand(() -> {
```
Remove commented out code

bot190 commented on src/main/java/frc/robot/subsystems/Indexer.java:
```diff
@@ -83,6 +85,13 @@ public CommandBase shootBall() {
         return cmd;
     }
+
+    public CommandBase waitTime() {
```
Is this used anywhere? It seems unnecessary to wrap `WaitCommand`.

msoucy commented on src/main/java/frc/robot/subsystems/Indexer.java:
```diff
@@ -83,6 +85,13 @@ public CommandBase shootBall() {
+
+    public CommandBase waitTime() {
+        CommandBase cmd = new WaitCommand(WAITTIME);
+        cmd.setName("Wait for tIME");
```
What's with the capitalization?

bot190 commented on src/main/java/frc/robot/subsystems/Drive.java:
```diff
@@ -132,4 +141,107 @@ public CommandBase turnDegrees(double degrees, double speed) {
+        return cmd;
+    }
+
+    // public CommandBase visionAdjust() {
```
Lets get rid of this command, there's no reason to keep the commented out code.

msoucy commented on src/main/java/frc/robot/subsystems/Lift.java:
```diff
@@ -119,12 +125,24 @@ public CommandBase turnOffBrake() {
+                    elevatorMotor.set(realSpeed);
+                } else if (ticks > 0 && lowerLimitSwitch.getAsBoolean()) {
+                    elevatorMotor.set(realSpeed);
+                }
```
Shouldn't there be an else here to stop the motor?

jeffaburke replied:
motor will never start

bot190 commented on src/main/java/frc/robot/subsystems/Drive.java:
```diff
@@ -132,4 +141,107 @@ public CommandBase turnDegrees(double degrees, double speed) {
+            pid.setSetpoint((SmartDashboard.getNumber("Angle Offset", 0)));
+            pid.setTolerance(2.5);
+            pid.setPID(0.025, 0.015, 0);
+            // System.out.println("second");
```
We can get rid of these commented lines as well.

bot190 commented on src/main/java/frc/robot/subsystems/Drive.java:
```diff
@@ -60,6 +64,7 @@ public Drive(DifferentialDriveMap map) {
         gyro = map.getGyro();
         driveTrain = new DifferentialDrive(leftMotorGroup, rightMotorGroup);
         driveTrain.setRightSideInverted(false);
+        pid = new PIDController(0, 0, 0);
```
We could probably put the PID values here instead of in the command.

Quinn-5 committed:
Merge remote-tracking branch 'origin/Connecticut' into Connecticut

bot190 commented on src/main/java/frc/robot/maps/RobotMap.java:
```diff
@@ -132,4 +134,15 @@ public IndexMap getIndexerMap() {
 
+    public static class LEDMap {
+        public AddressableLED visionLED() {
+            return new AddressableLED(-1);
```
This should be `0` instead of `-1` so it doesn't crash on other robots.

bot190 commented on src/main/java/frc/robot/maps/FrancoisMap.java:
```diff
@@ -222,4 +223,15 @@ public IEncoder getLiftEncoder() {
+        return new LEDMap() {
+            @Override
+            public AddressableLED visionLED() {
+                return new AddressableLED(0);
```
If we put this as the default we won't need to have any of this `LED` code here.

bot190 commented on src/main/java/frc/robot/Robot.java:
```diff
@@ -190,10 +200,23 @@ public CommandBase releaseBalls(int ballCount) {
+    }
+
+    public CommandBase fastReleaseBalls(int ballCount) {
+        CommandBase cmd = new SequentialCommandGroup(shooter.spinUp(5000), indexer.shootAllBalls(ballCount));
```
We'll confirm on Friday or so, but this likely needs to be a fair bit lower.

jeffaburke replied:
Sean was saying that we cat go all the way back in the trench and go in. Didn't think it was too low but he said we can't go all the back and make it in.

bot190 commented on src/main/java/frc/robot/Robot.java:
```diff
@@ -215,6 +238,29 @@ public CommandBase camToggle() {
         return camTogglecmd;
     }
 
+    public CommandBase camOn() {
```
This should probably be `enableTargeting`

bot190 commented on src/main/java/frc/robot/Robot.java:
```diff
@@ -215,6 +238,29 @@ public CommandBase camToggle() {
+        return cmd;
+    }
+
+    public CommandBase camOff() {
```
And this should probably be `disableTargeting`

bot190 reviewed (commented):
Some cleanup left to do.

jeffaburke committed:
FIxed Lift

jeffaburke committed:
Merge remote-tracking branch 'origin/Connecticut' into Connecticut

jeffaburke committed:
Pull Request Changes

didn't add factory command

bot190 commented on src/main/java/frc/robot/Robot.java:
```diff
@@ -66,9 +66,12 @@
 
     final private SendableChooser<Command> autoChooser = new SendableChooser<>();
 
+    final private int VisionGreen = 120;
```
This should not be here. It should be in the LED subsystem.

bot190 commented on src/main/java/frc/robot/maps/FrancoisMap.java:
```diff
@@ -222,4 +221,20 @@ public IEncoder getLiftEncoder() {
+
+            @Override
+            public AddressableLEDBuffer visionBuffer() {
+                return new AddressableLEDBuffer(6);
```
This should just return the number of LEDs not an actual buffer object.

bot190 commented on src/main/java/frc/robot/subsystems/ControlPanel.java:
```diff
@@ -106,7 +106,7 @@ public ColorStates getTargetColor() {
         CommandBase cmd = new FunctionalCommand(() -> {
         }, () -> {
-            spinnerMotor.set(speed.getAsDouble());
+            spinnerMotor.set(Math.abs(speed.getAsDouble()));
```
Maybe put a comment about why we're using `Math.abs` here.

bot190 commented on src/main/java/frc/robot/subsystems/Lift.java:
```diff
@@ -71,6 +71,13 @@ public CommandBase cancel() {
         return cmd;
     }
 
+    public CommandBase afterMatch() {
```
`resetLift` is a more descriptive name. Can you also add a comment explaining what this actually does?

bot190 commented on src/main/java/frc/robot/subsystems/Lift.java:
```diff
@@ -119,12 +125,13 @@ public CommandBase turnOffBrake() {
     }
 
-    public CommandBase moveTicks(double ticks, double speed) {
+    public CommandBase moveDisengage(double ticks, double speed) {
```
`moveForDisengage` would be better. We can probably also remove the ticks and speed arguments here since this is so specialized.

bot190 reviewed (commented):
Some final changes, but this is looking much better

Rohan-Sreenivasan committed:
Changes

jeffaburke committed:
Auto stash before merge of "Connecticut" and "origin/Connecticut"

jeffaburke committed:
PR Changes

jeffaburke committed:
Merge branch 'Connecticut2' into Connecticut

jeffaburke committed:
Oh no

jeffaburke committed:
Fixes for PR

jeffaburke committed:
Fixed Comment in Control Panel

Rohan-Sreenivasan committed:
Slow turn Boolean changes

Rohan-Sreenivasan committed:
Merge remote-tracking branch 'origin/Connecticut' into Connecticut

msoucy commented on src/main/java/frc/robot/maps/RobotMap.java:
```diff
@@ -132,4 +134,31 @@ public IndexMap getIndexerMap() {
+            return 0;
+        }
+
+        public ISolenoid visionRingLight() {
```
The lights are a solenoid? Change the name please.

bot190 replied:
Any suggestions for a name?

msoucy commented on src/main/java/frc/robot/subsystems/Drive.java:
```diff
@@ -132,4 +144,61 @@ public CommandBase turnDegrees(double degrees, double speed) {
+
+    public CommandBase slowTurn(boolean isTurningRight) {
+        CommandBase cmd = new RunCommand(() -> {
+            if (isTurningRight == true) {
```
Just use `if(isTurningRight) {}`. Using `== true` is redundant. Also below you can use the `else` without `else if`

andrewmartin5 replied:
resolved

msoucy commented on src/main/java/frc/robot/subsystems/Indexer.java:
```diff
@@ -115,17 +116,18 @@ public CommandBase reverseOutput() {
         CommandBase cmd = new FunctionalCommand(() -> {
         }, () -> {
-            if (frontIntakeIR.getAsBoolean()) {
+            if (frontIntakeIR.getAsBoolean() || backIntakeIR.getAsBoolean() && !topPierreIR.getAsBoolean()) {
```
is this `(a||b) && c`, or `a || (b && c)`? Use parenthesis.

andrewmartin5 replied:
resolved

msoucy commented on src/main/java/frc/robot/subsystems/Lift.java:
```diff
@@ -137,15 +143,39 @@ public CommandBase moveTicks(double ticks, double speed) {
         return cmd.withTimeout(0.2);
     }
 
+    public CommandBase moveTicks(double ticks, double speed) {
```
What is the distance of a "tick"? Please document.

bot190 replied:
We should rename this to distance.

andrewmartin5 replied:
resolvified

Rohan-Sreenivasan committed:
LEDS work

LED integration is much better and it works

msoucy commented on src/main/java/frc/robot/subsystems/Lift.java:
```diff
@@ -137,15 +143,39 @@ public CommandBase moveTicks(double ticks, double speed) {
+            elevatorMotor.stopMotor();
+        }, () -> {
+            return Math.abs(liftEncoder.getDistance()) >= Math.abs(ticks)
+                    || (ticks < 0 && upperLimitSwitch.getAsBoolean() || (ticks > 0 && lowerLimitSwitch.getAsBoolean()));
```
Since `ticks` doesn't change within this function call, you could do those checks before creating the functional command, and use just the appropriate limit switch here.

```java
BooleanSupplier limit = () -> true;
if (ticks < 0) {
  limit = upperLimitSwitch::getAsBoolean;
} else if (ticks > 0) {
  limit = lowerLimitSwitch::getAsBoolean;
}
// ...
return Math.abs() /* ... */ || limit.getAsBoolean();
```

andrewmartin5 replied:
resolvied

msoucy commented on src/main/java/frc/robot/subsystems/Lift.java:
```diff
@@ -137,15 +143,39 @@ public CommandBase moveTicks(double ticks, double speed) {
+        CommandBase cmd = new FunctionalCommand(() -> {
+            liftEncoder.reset();
+        }, () -> {
+            double realSpeed = speed;
```
This should be done once, outside of the functional command, instead of doing the math every cycle.

msoucy commented on src/main/java/frc/robot/subsystems/Lift.java:
```diff
@@ -137,15 +143,39 @@ public CommandBase moveTicks(double ticks, double speed) {
             }
         }, () -> {
-            return Math.abs(liftEncoder.getDistance()) >= ticks;
+            return Math.abs(liftEncoder.getDistance()) >= Math.abs(0.10);
```
Is the `abs` necessary? It's a constant... Also why isn't it a *named* constant?

andrewmartin5 replied:
pretty sure we don't quite need it. resolvification complete.

msoucy commented on src/main/java/frc/robot/subsystems/Led.java:
```diff
@@ -0,0 +1,74 @@
+
+    }
+
+    // public CommandBase visionLedColor(int hue, int saturation, int value) {
```
Commented code.

msoucy commented on src/main/java/frc/robot/subsystems/Drive.java:
```diff
@@ -60,6 +63,7 @@ public Drive(DifferentialDriveMap map) {
         gyro = map.getGyro();
         driveTrain = new DifferentialDrive(leftMotorGroup, rightMotorGroup);
         driveTrain.setRightSideInverted(false);
+        pid = new PIDController(0.025, 0.015, 0);
```
Does this need a tolerance?

msoucy commented on src/main/java/frc/robot/maps/FrancoisMap.java:
```diff
@@ -5,6 +5,7 @@
 import com.chopshop166.chopshoplib.RobotMapFor;
 import com.chopshop166.chopshoplib.maps.DifferentialDriveMap;
 import com.chopshop166.chopshoplib.outputs.ISolenoid;
+import com.chopshop166.chopshoplib.outputs.MockSolenoid;
```
Clean up your imports

msoucy commented on src/main/java/frc/robot/Robot.java:
```diff
@@ -215,6 +242,29 @@ public CommandBase camToggle() {
 
+    public CommandBase enableTargeting() {
+        CommandBase cmd = new InstantCommand(() -> {
+            SmartDashboard.putBoolean("Is Shooting", true);
```
If we use this variable a bunch, maybe use [getEntry](https://first.wpi.edu/FRC/roborio/release/docs/java/edu/wpi/first/wpilibj/smartdashboard/SmartDashboard.html#getEntry(java.lang.String)) and the entry's `getBoolean` and `setBoolean`

msoucy reviewed (changes requested):
😴 🔨

Quinn-5 committed:
make some changes for efficiency/move some functions to separate program

Quinn-5 committed:
Merge remote-tracking branch 'origin/Connecticut' into Connecticut

Rohan-Sreenivasan committed:
SHOOTER

shooter changes are good

Rohan-Sreenivasan committed:
Merge remote-tracking branch 'origin/Connecticut' into Connecticut

bot190 committed:
Small Shooter code cleanup

Remove unused command. 
Make shooterSpeed configure annotation work (hopefully).

bot190 committed:
Update to latest chopshoplib

This includes a helper class to de-bounce end conditions.

bot190 committed:
Threshold Check

bot190 committed:
Tune Vision based automation

This includes adjustments to the shooter speed calculations and vision alignment.

msoucy commented on src/main/java/frc/robot/subsystems/Shooter.java:
```diff
@@ -70,18 +83,71 @@ public CommandBase spinUp(final double speed) {
+            }
+
+            @Override
+            public void execute() {
```
Remove these if they're not used.

msoucy reviewed (changes requested):
There are some outstanding comments, plus this new one

bot190 committed:
BOB Testing

bot190 committed:
Limit shooter top speed calculation

The distance based shooter speed calculation is now limited to the physical top speed of the shooter. This allows us to actually shoot the ball instead of waiting forever to reach top speed.

bot190 committed:
Add PDP Motor mappings

This adds a CSV with the PDP motor mappings. It also adds a configuration file for DSLOG-Reader2 with a Francois profile.

Rohan-Sreenivasan committed:
Connecticut changes

Ill be back

jeffaburke committed:
Dont really know what this is just know i have to push

bot190 commented on src/main/java/frc/robot/subsystems/Lift.java:
```diff
@@ -137,15 +143,43 @@ public CommandBase moveTicks(double ticks, double speed) {
+        } else if (ticks > 0) {
+            limittmp = lowerLimitSwitch::getAsBoolean;
+        }
+        final BooleanSupplier limit = limittmp;
```
Why not just use `limittmp` directly?

msoucy replied:
It has to be `final` because of silly Java rules

bot190 commented on src/main/java/frc/robot/subsystems/Shooter.java:
```diff
@@ -33,16 +35,22 @@
     public final static double BALL_SPEED_RATIO = 27.358;
 
+    private final static double MAX_SHOOTER_SPEED = 5400;
+    // y = 2878.7x^(0.3094)
```
We can probably get rid of this comment now.

bot190 commented on src/main/java/frc/robot/subsystems/Shooter.java:
```diff
@@ -70,18 +83,75 @@ public CommandBase spinUp(final double speed) {
             // TODO incorperate calculations
             // shooterWheelMotor.set(calculateRPM() / MAX_RPM);
-            shooterWheelMotor.setSetpoint(speed);
+            shooterWheelMotor.setSetpoint(shooterSpeed);
```
I've co-opted this function to use a variable on the dashboard. We should probably make this a new command.

bot190 commented on src/main/java/frc/robot/subsystems/Shooter.java:
```diff
@@ -70,18 +83,75 @@ public CommandBase spinUp(final double speed) {
+            double dist = SmartDashboard.getNumber("Distance To Target", 3.8);
+            // If we don't see the target then default to the Initiation line speed
+            if (!SmartDashboard.getBoolean("Sees Target", false)) {
+                output = 4700;
```
This should become a `final` value.

jeffaburke replied:
```
        if (!SmartDashboard.getBoolean("Sees Target", false)) {
            output = 4700;
        }
        output = ((119.1 * (Math.pow(dist, 2))) - (1000 * dist) + 6300);
```
Really? I don't think it should

jeffaburke replied:
actually should there be an else there?

bot190 commented on src/main/java/frc/robot/subsystems/Shooter.java:
```diff
@@ -70,18 +83,75 @@ public CommandBase spinUp(final double speed) {
         return cmd;
     }
 
+    public CommandBase linearSpeedUp() {
```
This should also have the `ThresholdCheck`

bot190 commented on src/vision/Offline_Filter.py:
```diff
@@ -1,141 +1,154 @@
+LINES = cv2.HoughLinesP(MED_EDGES, 1, radians(0.5), 25, maxLineGap=25)
+
+# If there are lines:
+if LINES is not None:
```
All of this line matching code should be in a helper function inside `visionfunctions.py`

bot190 commented on src/vision/frc_vision_final.py:
```diff
@@ -81,96 +49,84 @@ def newLine(
         # If there are lines:
-        if LINES is not None:
-            NUM_LINES = len(LINES)
+        if lines is not None:
```
Instead, this should be 
```
if lines is None:
	sd.putBoolean("Sees Target", False)
	continue
```
This would remove a lot of the indentation.

bot190 reviewed (commented):
The vision code should probably be re-written to be more modular.

bot190 committed:
Removed configure() call that doesn't work

I had added this configure function to hopefully get the vision code to connect to the robot, but it breaks the python code.

andrewmartin5 committed:
lift code pr fixes

not fully fixed. there are more changes required.

andrewmartin5 committed:
Merge remote-tracking branch 'origin/Connecticut' into Connecticut

msoucy commented on src/main/java/frc/robot/subsystems/Led.java:
```diff
@@ -0,0 +1,60 @@
+    AddressableLED liftLED;
+    ISolenoid ringLight;
+    AddressableLEDBuffer liftBuffer;
+    double firstHue;
```
Where is this used?

msoucy commented on src/main/java/frc/robot/subsystems/Led.java:
```diff
@@ -0,0 +1,60 @@
+        return cmd;
+    }
+
+    // public CommandBase visionLedStream() {
```
Commented out code

msoucy commented on src/main/java/frc/robot/subsystems/Intake.java:
```diff
@@ -72,4 +72,11 @@ public CommandBase deployIntake() {
     }
+
+    public CommandBase retractIntake() {
+        return new InstantCommand(() -> {
```
I'd like this command to be named

msoucy commented on src/main/java/frc/robot/subsystems/Shooter.java:
```diff
@@ -70,18 +83,75 @@ public CommandBase spinUp(final double speed) {
+            if (!SmartDashboard.getBoolean("Sees Target", false)) {
+                output = 4700;
+            }
+            output = ((119.1 * (Math.pow(dist, 2))) - (1000 * dist) + 6300);
```
This needs to be in an `else`, otherwise it'll override the last value.

msoucy commented on src/vision/frc_vision_final.py:
```diff
@@ -95,82 +73,56 @@ def newLine(
-            FILTERED_LINE_IMG = np.zeros((HEIGHT, WIDTH, 3), np.uint8)
-            LINE_IMG = np.zeros((HEIGHT, WIDTH, 3), np.uint8)
+        isShooting = sd.getBoolean("Is Shooting", defaultValue=False)
+        if not isShooting == OldStream:
```
`if isShooting != OldStream:`

jeffaburke committed:
Connecticut changes

msoucy commented on src/vision/Offline_Filter.py:
```diff
@@ -1,141 +1,154 @@
-# PyLint please stop yelling at me I did nothing wrong...
-from math import radians, degrees
+# Final FRC 2020 vision program
```
How is this different from the "final" file? If it does the same logic, it should be moved into a function and then into a shared file.

msoucy commented on src/vision/frc_vision_final.py:
```diff
@@ -95,82 +73,56 @@ def newLine(
                 if FILTERED_LINES:
                     if (
-                        new_slope < -ANGLE_THRESHOLD or new_slope > ANGLE_THRESHOLD
+                        ANGLE_HIGH_THRESHOLD > abs(new_slope) > ANGLE_LOW_THRESHOLD
```
Instead of calling a function within that block, set a variable. Then do `if variable: newLine()`.

Then remove the `newLine` function and just replace it with the contents, since it's more problematic than useful. It takes nine parameters...

Rohan-Sreenivasan committed:
Shooter final speed

Rohan-Sreenivasan committed:
Its better now

Rohan-Sreenivasan committed:
Indexer changes

Rohan-Sreenivasan committed:
Shooter change

Rohan-Sreenivasan committed:
Led changes

Rohan-Sreenivasan committed:
comment deleted

msoucy reviewed (dismissed):
Vision can be worked on in a different PR.

Rohan-Sreenivasan committed:
Led changes, even more

Rohan-Sreenivasan committed:
Lift led rename

Rohan-Sreenivasan committed:
Buffer

bot190 reviewed (commented):
Looks good to me.

msoucy reviewed (approved):
🚢 it
